include/RED4ext/SystemUpdate.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <cstdint>
+
+#include <RED4ext/Callback.hpp>
+#include <RED4ext/Common.hpp>
+#include <RED4ext/JobQueue.hpp>
+
+namespace RED4ext
+{
+struct IUpdatableSystem;
+
+enum class UpdateTickGroup : uint8_t
+{
+    FrameBegin,
+    Multiplayer_UpdateStateSnapshots,
+    EntityUpdateState,
+    PreBuckets,
+    Buckets,
+    PostBuckets,
+    CameraUpdate,
+    PlayerAimUpdate,
+    PostPlayerAimUpdate,
+    MappinsUpdate,
+    BlackboardCallbacks_SecondPass,
+    PreRenderUpdate,
+    Multiplayer_CaptureStateSnapshots,
+    Unknown,
+};
+
+enum class UpdateBucketEnum : uint8_t
+{
+    Vehicle = 0,
+    Character = 1,
+    AttachedObject = 2,
+};
+
+enum class UpdateBucketMask : uint8_t
+{
+    Vehicle = 1,
+    Character = 2,
+    AttachedObject = 4,
+    Everything = 7,
```

Comment:
```suggestion
    Vehicle = 1 << 0,
    Character = 1 << 1,
    AttachedObject = 1 << 2,
    Everything = 7,
```

Reply:
Ultimately it can be:
```cpp
enum class UpdateBucketMask : uint8_t
{
    Vehicle = 1 << static_cast<uint8_t>(UpdateBucketEnum::Vehicle),
    Character = 1 << static_cast<uint8_t>(UpdateBucketEnum::Character),
    AttachedObject = 1 << static_cast<uint8_t>(UpdateBucketEnum::AttachedObject),
    Everything = Vehicle | Character | AttachedObject,
};
```

Reply:
Even better!

Reply:
Done.